### PR TITLE
Make multi-threaded shuffle not experimental and update docs

### PR DESCRIPTION
**Pull request by abellina**
Closes https://github.com/NVIDIA/spark-rapids/issues/6807.

We have used the multi-threaded shuffle for a while in our performance cluster, and we gained more confidence in this mode of operation, and it is great that this mode doesn't require special hardware or special system libraries installed. This updates the documentation to reflect that, and removes the "experimental" label.

### CODE REVIEW

**Comment by abellina** [comment]
build

File: docs/additional-functionality/rapids-shuffle.md [hunk 1]
```diff
@@ -8,13 +8,50 @@ nav_order: 5
...
+configuration can be independently changed for writers and readers using:
+`spark.rapids.shuffle.multiThreaded.[writer|reader].threads`. An appropriate value for these
+pools is the number of cores in the system divided by the number of executors per machine.
 
 ## UCX Mode
 
-UCX mode is the default for the RAPIDS Shuffle Manager. It has two components: a spillable cache, 
-and a transport that can utilize Remote Direct Memory Access (RDMA) and high-bandwidth transfers 
-within a node that has multiple GPUs. This is possible because the plugin 
+UCX mode has two components: a spillable cache, and a transport that can utilize Remote Direct Memory Access (RDMA) and high-bandwidth transfers
```
Comment (tgravescs) [inline docs/additional-functionality/rapids-shuffle.md:53]: nit,line length

File: docs/additional-functionality/rapids-shuffle.md [hunk 2]
```diff
@@ -8,13 +8,50 @@ nav_order: 5
...
+This mode is similar to the built-in Spark shuffle, but it attempts to use more CPU threads
+for compute-intensive tasks, such as compression and decompression. 
+
+Minimum configuration:
+
+```shell
+--conf spark.shuffle.manager=com.nvidia.spark.rapids.[shim package].RapidsShuffleManager \
+--conf spark.driver.extraClassPath=${SPARK_RAPIDS_PLUGIN_JAR} \
+--conf spark.executor.extraClassPath=${SPARK_RAPIDS_PLUGIN_JAR} \
+--conf spark.rapids.shuffle.mode=MULTITHREADED
```
Comment (tgravescs) [inline docs/additional-functionality/rapids-shuffle.md:43]: we don't need this if its default now right?

Comment (abellina) [reply docs/additional-functionality/rapids-shuffle.md:43]: whoops right :)

**Comment by abellina** [comment]
@tgravescs addressed your comments here: https://github.com/NVIDIA/spark-rapids/pull/7317/commits/f2ea21eaf7a2158111d5e34a1b78ec01ae01083f

**Comment by abellina** [comment]
build

**Comment by abellina** [comment]
build

**Comment by abellina** [comment]
build

File: docs/additional-functionality/rapids-shuffle.md [hunk 3]
```diff
@@ -8,13 +8,50 @@ nav_order: 5
...
 
 ## UCX Mode
 
-UCX mode is the default for the RAPIDS Shuffle Manager. It has two components: a spillable cache, 
-and a transport that can utilize Remote Direct Memory Access (RDMA) and high-bandwidth transfers 
-within a node that has multiple GPUs. This is possible because the plugin 
+UCX mode has two components: a spillable cache, and a transport that can utilize 
+Remote Direct Memory Access (RDMA) and high-bandwidth transfers 
+within a node that has multiple GPUs. This is possible because this mode 
 utilizes [Unified Communication X (UCX)](https://www.openucx.org/) as its transport.
```
Comment (tgravescs) [inline docs/additional-functionality/rapids-shuffle.md:55]: do we want to mention setting  spark.rapids.shuffle.mode to ucx?

Comment (abellina) [reply docs/additional-functionality/rapids-shuffle.md:55]: yeap, sorry. Great catch

Comment (abellina) [reply docs/additional-functionality/rapids-shuffle.md:55]: Ok this should be fixed now.

**Review by tgravescs — APPROVED** [review]

**Comment by jlowe** [comment]
build

**Pull request merged**